### DISCUSSION TITLE
Do not use json payload but split items over topics so clients can select data they want

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#11069] [Ideas]
At this moment STATE and INFO for example is one json payload containing all state information.

I think it is more versatile to split the json data into different topics. This way a mqtt client can receive exactly the information it is interested in and the json doesn't have to be parsed anymore before processing in for example nodered.

I know this is a very big change, and maybe it has to be an additional feature.

Example:
home/TASMOTA_ZIGBEE/tele/INFO1 = {"Module":"Sonoff ZbBridge","Version":"9.3.0(zbbridge)","FallbackTopic":"cmnd/DVES_9BA274_fb/","GroupTopic":"home/tasmotas/cmnd/"}

Becomes:
home/TASMOTA_ZIGBEE/tele/INFO/module = "Sonoff ZbBridge"
home/TASMOTA_ZIGBEE/tele/INFO/version = "9.3.0(zbbridge)"
home/TASMOTA_ZIGBEE/tele/INFO/fallbacktopic = "cmnd/DVES_9BA274_fb/"
home/TASMOTA_ZIGBEE/tele/INFO/grouptopic = "home/tasmotas/cmnd/"

**Comment by barbudor** [comment]
JSON is a convenient format so that one's can pickup what it needs
Non-Json message are possible for non complex responses like Power
There's zero chance this would be acepted

**Comment by ascillato2** [comment]
Thanks for sharing your ideas. Most automation softwares like OpenHab, Homeassistant, etc, parse JSON messages without issues. Sorry, this change don't have any advantage for most users and make a more intensive use of MQTT Broker. Sorry, that is not going to be added.

**Reply by sfromis** [reply]
I'd say that atomizing the payloads that way would be *less* versatile than the current format, which is well-aligned with the very widespread JSON format where common tools have easy parsing. Quite a bit easier than having to collect the data from several messages.

It would be quite easy to write a small bridge atomizing the payloads in tools like Node-Red.

**Comment by jeroenst** [comment]
Thanx for sharing your visions.

It's not a big problem to me but I just wanted to share my ideas and to hear how others think about it.

**Comment by jeroenst** [comment]
Because values aren't separated into different topics I bump into this issue: https://github.com/arendst/Tasmota/issues/11103
